packages/api/src/router/post.ts

```diff
@@ -6,4 +8,42 @@ export const postRouter = createTRPCRouter({
       orderBy: { createdAt: "desc" },
     });
   }),
+  createPost: protectedProcedure
+    .input(z.object({ title: z.string(), contents: z.string() }))
+    .mutation(async ({ ctx, input }) => {
+      await ctx.db.post.create({
+        data: {
+          title: input.title,
+          contents: input.contents,
+        },
+      });
+    }),
+  updatePostByID: protectedProcedure
+    .input(
+      z.object({
+        id: z.number(),
+        title: z.string().optional(),
+        contents: z.string().optional(),
+      }),
+    )
+    .mutation(async ({ ctx, input }) => {
+      await ctx.db.post.update({
+        where: {
+          id: input.id,
+        },
+        data: {
+          title: input.title,
+          contents: input.contents,
+        },
```

Comment:
did you check that this doesn't change the value if the value isn't provided in the update?

Reply:
Yes, fields are not updated if input is undefined/left empty for one